### ISSUE TITLE
`Unused method argument warning` is thrown when argument used in block

### ISSUE THREAD

**Issue by dimus** [bbatsov/rubocop#1864]
The following code generates warning:
```
W: Unused method argument - id. If it's necessary, use _ or _id as an argument 
name to indicate that it won't be used. You can also write as find(*) if you want the 
method to accept any arguments but don't care about them.
```
```ruby
def self.find(id)
  fetch.find(&:id)
end
```
Rubocop version 0.30.1

**Comment by jonas054** [comment]
The given code is equivalent to
```ruby
def self.find(id)
  fetch.find { |elem| elem.id }
end
```
so the `id` argument is, in fact, not used. Or am I missing something?

**Comment by bbatsov** [comment]
You're not missing anything.

**Comment by dimus** [comment]
my bad, aplogies
